### PR TITLE
Adding a code sample of how the modal widget can be initialized

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this PR

This PR is an improvement of Modal widget, which provides an example of how the modal can be used and how the widget's options can be passed during initialization. Also the example's result preview is provided.
Moreover, an additional (important) widget's option was added to the documentation.

## Affected URLs

- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_modal.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_modal.html
- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html

whatsnew
Added a [new code sample](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html#code-sample) showing how to initialize a modal widget and pass options to it.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by rogyar** [comment]
CodeBase: https://github.com/magento/magento2/blob/2.1/app/code/Magento/Ui/view/base/web/js/modal/modal.js#L73

File: guides/v2.1/javascript-dev-guide/widgets/widget_modal.md [hunk 1]
```diff
@@ -202,6 +209,36 @@ Called when the modal window is opened.
...
+        Popup Content
+    </div>
+</div>
+```
+
+### Result
+
+The result is a modal and a button (_Click Here_) which opens the modal.
+
+![Tabs Widget]({{ page.baseurl }}/javascript-dev-guide/widgets/images/modal-widget-result.png)
```
Comment (rogyar) [inline guides/v2.1/javascript-dev-guide/widgets/widget_modal.md]: Please, double check the `[Tabs Widget]` title here. The picture is about a modal widget example.

File: guides/v2.1/javascript-dev-guide/widgets/widget_modal.md [hunk 2]
```diff
@@ -138,6 +139,12 @@ Translated title for popup window.
 
 **Default value**: empty
 
+### `trigger` {#modal_trigger}
+The element which triggers the modal.
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_modal.md]:
```suggestion
The element that triggers the modal.
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_modal.md [hunk 3]
```diff
@@ -202,6 +209,36 @@ Called when the modal window is opened.
...
+        }}">
+    <div class="content">
+        Popup Content
+    </div>
+</div>
+```
+
+### Result
+
+The result is a modal and a button (_Click Here_) which opens the modal.
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_modal.md]:
```suggestion
The result is a modal and a button (_Click Here_) that opens the modal.
```

**Comment by eduard13** [comment]
Hi @jeff-matthews, the suggested changes were applied.
Thank you for the review.

**Comment by eduard13** [comment]
@jeff-matthews, yes they are. Interesting, I didn't know about that, thank you for noticing.
Please note that I moved it into the mentioned folder, optimized it and removed the other ones.
Thank you.

**Comment by jeff-matthews** [comment]
Perfect @eduard13! Thank you for taking the extra step to optimize the image as well. We haven't had good contributor guidelines around doing that in the past, but we'll create some soon. Thank you for being proactive.

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Tests failed. Here's the log:

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "LinkCheck", "ScriptCheck"] on ["_site"] on *.html... 

Ran on 3824 files!

- internal image /common/images/js/widgets/modal-widget-result.png does not exist
  *  _site/guides/v2.1/javascript-dev-guide/widgets/widget_modal.html (line 2368)
  *  _site/guides/v2.2/javascript-dev-guide/widgets/widget_modal.html (line 2430)
  *  _site/guides/v2.3/javascript-dev-guide/widgets/widget_modal.html (line 2486)
rake aborted!
HTML-Proofer found 3 failures!```

File: guides/v2.1/javascript-dev-guide/widgets/widget_modal.md [hunk 4]
```diff
@@ -202,6 +209,36 @@ Called when the modal window is opened.
...
+        Popup Content
+    </div>
+</div>
+```
+
+### Result
+
+The result is a modal and a button (_Click Here_) that opens the modal.
+
+![Modal Widget]({{ site.baseurl }}/common/images/js/widgets/modal-widget-result.png)
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_modal.md]:
```suggestion
![Modal Widget]({{ site.baseurl }}/common/images/js/widget/modal-widget-result.png)
```

Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_modal.md]:
```suggestion
![Modal Widget]({{ site.baseurl }}/common/images/js/widget/modal-widget-result.png)
```

**Comment by jeff-matthews** [comment]
running tests

**Comment by eduard13** [comment]
@jeff-matthews now it should be fine I guess.

**Comment by jeff-matthews** [comment]
Yes, I think so too. Almost finished running tests.

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**